### PR TITLE
Docker Hub Sample images

### PR DESCRIPTION
**Pull request by WillAyd**
Still working on mamba. Opened an open source account inquiry with Docker so will eventually place there. In the interim can place on personal account for testing

### CODE REVIEW

**Comment by phofl** [comment]
Just to understand: This is for development with different environment sizes?

**Comment by WillAyd** [comment]
Yea it could help target different development goals. Alpine would be a super small image (~300 MB total), the minimal variants for pip/mamba just give you everything you need to compile / test most of the repository and come in at ~1 GB. The full variants would provide all packages for a complete environment, at the cost of more disk space

I put the minimal images up at https://hub.docker.com/r/willayd/pandas-dev/tags if you'd like to try

**Comment by WillAyd** [comment]
Not a goal at the moment but there *might* also be a future where instead of rebuilding a testing image for every CI run we have a pre-built docker image that we just download and run from. Might save a lot of build time

File: ci/dockerfiles/Dockerfile.pip-minimal [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.10.8
```
Comment (mroeschke) [inline ci/dockerfiles/Dockerfile.pip-minimal:1]: Could we use slim?

File: ci/dockerfiles/Dockerfile.alpine [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM python:3.10.8-alpine
+
+RUN apk update & apk upgrade
+RUN apk add build-base
+
+RUN python -m pip install \
+  cython \
```
Comment (mroeschke) [inline ci/dockerfiles/Dockerfile.alpine:7]: Could we maintain the minimal set with files?

**Comment by mroeschke** [comment]
I see that the pip vs mamba minimals are pretty close in size while the mamba ones have a ~1G difference.

Was the thought of providing pip vs mamba due to a size difference? If the size difference isn't significant, I would slightly favor just having the mamba environments, especially this is for scientific python development in which mamba/conda is well more equipped than pip

**Comment by WillAyd** [comment]
Is that the size you see after installing locally? Might have some artifacts left over locally that I'm not aware of but the mamba full install for me was almost double the size of the pip image. 

The mamba image is also a little more complicated to deal with if you want to run pre commit. Not sure if that would even work but the pip images should

**Comment by mroeschke** [comment]
Oh that's just the compressed sizes I'm seeing in https://hub.docker.com/r/willayd/pandas-dev/tags

**Comment by WillAyd** [comment]
Ah gotcha. Yea would be curious if anyone does `docker pull willayd/pandas-dev:mamba-all` and `docker pull willayd/pandas-dev:pip-all` if you see a difference. Here's what mine yield

```sh
$ docker image list
REPOSITORY                      TAG                      IMAGE ID       CREATED         SIZE
willayd/pandas-dev              mamba-all                6336333277b7   2 hours ago     5.68GB
willayd/pandas-dev              pip-all                  b209d2a0b20b   2 days ago      2.63GB
```

There could be something simple missing from the mamba Dockerfile that would minimize that, but I think it will always use more disk space than pip

File: ci/dockerfiles/Dockerfile.alpine (hunk 2 above)
Comment (WillAyd) [reply ci/dockerfiles/Dockerfile.alpine:7]: Absolutely. One option is to create a `requirements-dev-minimal.txt` file in this CI folder, then can just execute a `COPY` command within the Dockerfile and install from that

File: ci/dockerfiles/Dockerfile.pip-minimal (hunk 1 above)
Comment (WillAyd) [reply ci/dockerfiles/Dockerfile.pip-minimal:1]: Oh nice idea. Would have to do a bit more research though - the slim packages don't include compilers. Could get them with `apt install build-essential` but that adds 312 MB to the image size, so defeats the purpose a bit

Comment (WillAyd) [reply ci/dockerfiles/Dockerfile.pip-minimal:1]: OK this was an awesome idea. So if you just install gcc and g++ you get from the current 1.06 GB image size down to 475 MB

File: ci/dockerfiles/Dockerfile.mamba-minimal [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM quay.io/condaforge/mambaforge
+
+RUN apt update && apt upgrade -y
+RUN DEBIAN_FRONTEND=noninteractive apt install -y tzdata
+
+RUN mamba create -n pandas-dev \
```
Comment (WillAyd) [inline ci/dockerfiles/Dockerfile.mamba-minimal:6]: This is disconnected from requirements-minimal.txt at the moment. Could create a separate env file, break this up into two steps with a pip install, or leave as is

File: ci/dockerfiles/requirements-minimal.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+cython
```
Comment (WillAyd) [inline ci/dockerfiles/requirements-minimal.txt:1]: This file has some overlap with #50339

**Comment by WillAyd** [comment]
We now an official pandas org set up on docker hub - updated the documentation to point to that

**Comment by simonjayhawkins** [comment]
@WillAyd still active? also a conflict needs resolving.

**Pull request closed without merging**